app/assets/javascripts/float-kit/addon/lib/d-toast-instance.js

```diff
@@ -53,4 +55,9 @@ export default class DToastInstance {
 
     cancel(this.autoCloseHandler);
   }
+
+  @action
+  resumeAutoClose(event) {
+    this.registerAutoClose(event.target);
```

Comment:
I suppose it means that the modifier teardown function will not be called. But in this case it's *kinda* ok because the cleanup from the original invocation will still run? But it still kinda feels like we're misusing modifiers here.

If we want to make thing more declarative, while keeping the modifier tightly-coupled to the component, we could do something like this:

```hbs
<output
  {{(if autocloseActive (modifier registerAutoClose))}}
/>
```

and then js would be something like

```js
@tracked autocloseActive = true;

registerAutoClose = modifier(... // as before

@action
cancelAutoClose(){
  this.autocloseActive = false;
}

@action
resumeAutoClose(){
  this.autocloseActive = true;
}

```


Or we could separate out the modifier so that it sets up the mouseenter/mouseleave listeners itself, and everything is nicely separated from the component.

```js
class AutoHideToast extends Modifier {
  constructor(owner, args) {
    super(owner, args);
    registerDestructor(this, (instance) => instance.cleanup());
  }

  modify(element, _, { duration }){
    this.element = element;
    this.duration = duration;
    element.addEventListener("mouseenter", this.stopTheTimer);
    element.addEventListener("mouseleave", this.startTheTimer);
    this.startTheTimer();
  }


  cleanup(){
    // stop timer and remove event listeners
  }
}
```

This feels cleaner to me because the modifier won't have access to the component's internal state. It would simply be used like


```
<output
  {{(if toast.options.autoclose (modifier AutoHideToast duration=toast.options.duration))}}
/>

Reply:
Yes this is what I have done at other places (using tracked properties to trigger modifiers), like the discourse-ai animation in topics. Ok thanks :+1: